mlrun/common/secrets.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from abc import ABC, abstractmethod
+
+import mlrun.common.schemas
+
+
+class SecretProviderInterface(ABC):
+    @abstractmethod
+    def store_auth_secret(
+        self, username: str, access_key: str, namespace=""
+    ) -> (str, mlrun.common.schemas.SecretEventActions):
+        pass
+
+    @abstractmethod
+    def delete_auth_secret(self, secret_ref: str, namespace=""):
+        pass
+
+    @abstractmethod
+    def read_auth_secret(self, secret_name, namespace="", raise_on_not_found=False):
+        pass
+
+    @abstractmethod
+    def store_project_secrets(
+        self, project, secrets, namespace=""
+    ) -> (str, mlrun.common.schemas.SecretEventActions):
+        pass
+
+    @abstractmethod
+    def delete_project_secrets(self, project, secrets, namespace=""):
+        pass
+
+    @abstractmethod
+    def get_project_secret_keys(self, project, namespace="", filter_internal=False):
+        pass
+
+    @abstractmethod
+    def get_project_secret_data(self, project, secret_keys=None, namespace=""):
+        pass
+
+    @abstractmethod
+    def get_secret_data(self, secret_name, namespace=""):
+        pass
+
+
+class InMemorySecretProvider(SecretProviderInterface):
+    def __init__(self):
+        # project -> secret_key -> secret_value
+        self.project_secrets_map = {}
+        # ref -> secret_key -> secret_value
+        self.auth_secrets_map = {}
+        # secret-name -> secret_key -> secret_value
+        self.secrets_map = {}
+
+    @staticmethod
+    def get_auth_secret_name(username: str, access_key: str) -> str:
+        return f"secret-ref-{username}-{access_key}"
```

Comment:
```suggestion
    def resolve_auth_secret_name(username: str, access_key: str) -> str:
        return f"secret-ref-{username}-{access_key}"
```

use get _when_ you go and get, like literally. here you just ... resolve/compile it

Reply:
also, better of keeping interface functionality  first